sklearn/ensemble/_hist_gradient_boosting/_predictor.pyx

```diff
@@ -21,46 +21,162 @@ from .common cimport node_struct
 np.import_array()
 
 
+@cython.final
+cdef class TreePredictor:
+    """Tree class used for predictions.
+
+    Parameters
+    ----------
+    nodes_cnt : int
+        Number of nodes in tree.
+    """
+    def __init__(self, int node_cnt):
+        self._nodes.resize(node_cnt)
+
+    cdef node_struct* get(self, int node_idx) nogil:
+        """Get reference to node."""
+        return &self._nodes[node_idx]
+
+    cdef int get_size(self) nogil:
+        """Get number of nodes in tree."""
+        return self._nodes.size()
+
+    @property
+    def nodes(self):
+        """Return nodes as a list of dicts."""
+        return self._nodes
+
+    def get_n_leaf_nodes(self):
+        """Return number of leaves."""
+        cdef:
+            int n_leaf_nodes = 0
+            node_struct node
+
+        for node in self._nodes:
+            n_leaf_nodes += node.is_leaf
+        return n_leaf_nodes
+
+    def get_max_depth(self):
+        """Return maximum depth among all leaves."""
+        cdef:
+            int max_depth = 0
+            node_struct node
+
+        for node in self._nodes:
+            if node.depth > max_depth:
+                max_depth = node.depth
+        return max_depth
+
+    def predict(self, X):
+        """Predict raw values for non-binned data.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_samples, n_features)
+            The input samples.
+
+        Returns
+        -------
+        y : ndarray, shape (n_samples,)
+            The raw predicted values.
+        """
+        out = np.empty(X.shape[0], dtype=Y_DTYPE)
+        _predict_from_numeric_data(self, X, out)
+        return out
+
+    def predict_binned(self, X, missing_values_bin_idx):
+        """Predict raw values for binned data.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_samples, n_features)
+            The input samples.
+        missing_values_bin_idx : uint8
+            Index of the bin that is used for missing values. This is the
+            index of the last bin and is always equal to max_bins (as passed
+            to the GBDT classes), or equivalently to n_bins - 1.
+
+        Returns
+        -------
+        y : ndarray, shape (n_samples,)
+            The raw predicted values.
+        """
+        out = np.empty(X.shape[0], dtype=Y_DTYPE)
+        _predict_from_binned_data(self, X, missing_values_bin_idx, out)
+        return out
+
+    def compute_partial_dependence(self, grid, target_features, out):
```

Comment:
same here

Reply:
Let's just copy the content of `_compute_partial_dependence()` here? We don't need this wrapper anymore.